gateway-js/src/loadSupergraphSdlFromStorage.ts

```diff
@@ -38,6 +38,57 @@ const { name, version } = require('../package.json');
 
 const fetchErrorMsg = "An error occurred while fetching your schema from Apollo: ";
 
+var fetchCounter: number = 0;
+
+export class SupergraphSdlResult {
+  id: string;
+  supergraphSdl: string;
+  constructor(id: string, supergraphSdl: string) {
+    this.id = id;
+    this.supergraphSdl = supergraphSdl;
+  }
+}
+
+export async function loadSupergraphSdlFromUplinks({
+  graphRef,
+  apiKey,
+  endpoints,
+  errorReportingEndpoint,
+  fetcher,
+  compositionId,
+  maxRetries,
+}: {
+  graphRef: string;
+  apiKey: string;
+  endpoints: string[];
+  errorReportingEndpoint: string | undefined,
+  fetcher: typeof fetch;
+  compositionId: string | null;
+  maxRetries: number
+}) : Promise<SupergraphSdlResult | null> {
+  var retries = 0;
+  var lastException = null;
+  var result: SupergraphSdlResult | null = null;
+  while (retries++ <= maxRetries && result == null) {
+    try {
+      result = await loadSupergraphSdlFromStorage({
+        graphRef,
+        apiKey,
+        endpoint: endpoints[fetchCounter++ % endpoints.length],
+        errorReportingEndpoint,
+        fetcher,
+        compositionId
+      });
+    } catch (e) {
+      lastException = e;
+    }
+  }
+  if (result == null && lastException != null) {
+    throw lastException;
+  }
```

Comment:
In JS, both `null == null` and `undefined == null` return true. As such, `== null` carries a semantic _implication_ that we might expect null or undefined on the left hand side (in my examples).

Since we've established the types ourselves to either be something truthy or `null`, we can use the `===` and `!==` instead to be more clear about our expectations of the possible values here.

Reply:
I now use `===` and `!==` here.